pkg/config/config.go

```diff
@@ -258,7 +258,19 @@ func GetIngestPort() uint64 {
 // returns the configured query port
 // if the node is not a query node, this will not be set
 func GetQueryPort() uint64 {
-	return runningConfig.QueryPort
+	var port uint64
+	value := os.Getenv("PORT")
+	if value != "" {
+		n, err := strconv.Atoi(value)
+		if err != nil {
+			log.Error("Error while converting port to int")
+			n = 80
+		}
+		port = uint64(n)
```

Comment:
In this error case where `PORT` is defined but not a number, we can just fall back to `runningConfig.QueryPort` instead of using `80` as the default.